### DISCUSSION TITLE
extracting the contents of eNVM and eMMC devices on the Polar Fire SoC Video kit

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#317] [Reference designs, FPGA and development kits]
Hi,
customer just received a few of these kits and would like to save the contents of eNVM and eMMC that comes by default on the board.   They plan on doing some bootloader work and updates to one or both of these is very likely.
I guess the question is how can one extract the contents or maybe how can one bring the board back to it's original state, maybe "unbrick" is the right term.  
We can make the kit a USB mass storage device, usbmmc command ,  that gives us a way to update the eMMC I believe.   
Has anyone gone down this path before?  
Thx!
Dave

**Comment by hughbreslin** [comment]
I think trying to extract and directly reprogram the content of those memories will be very difficult... Can they not just update to the latest reference design and Linux image by following the [docs](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/reference-designs-fpga-and-development-kits/icicle-kit-user-guide.md#updating) and if they need to revert they'll have a copy of the FPExpress job and .wic from the tag they've chosen?
